### PR TITLE
feat: Create EvmIcaModule

### PR DESCRIPTION
**Pull request by ltyu**
### Description
Adds an ICA Module with `create()` to be used in an upcoming CoreModule PR. For now the `read()` and `update()` are not implemented because its unclear if they will be used in CoreModule

### Backward compatibility

Yes

### Testing

Unit Tests

### CODE REVIEW

**Comment by ltyu** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/hyperlane-xyz/hyperlane-monorepo/3729?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3729** <a href="https://app.graphite.dev/github/pr/hyperlane-xyz/hyperlane-monorepo/3729?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3681** <a href="https://app.graphite.dev/github/pr/hyperlane-xyz/hyperlane-monorepo/3681?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ltyu and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: typescript/sdk/src/core/EvmIcaModule.hardhat-test.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          mailbox: mailbox.address,
+          owner: signer.address,
+        },
+        multiProvider,
+      });
+
+      const { interchainAccountRouter, interchainAccountIsm } =
+        evmIcaModule.serialize();
+      expect(interchainAccountIsm).to.exist;
+      expect(interchainAccountRouter).to.exist;
```
Comment (jmrossy) [inline typescript/sdk/src/core/EvmIcaModule.hardhat-test.ts:38]: Might be good to get more specific checks here

**Review by jmrossy — APPROVED** [review]
A good start!

**Review by paulbalaji — APPROVED** [review]

File: typescript/sdk/src/core/EvmIcaModule.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import { ChainNameOrId } from '../types.js';
+
+import { HyperlaneModule, HyperlaneModuleArgs } from './AbstractHyperlaneModule.js';
+
+export type InterchainAccountConfig = ProxiedRouterConfig;
+
+export class EvmIcaModule extends HyperlaneModule<
+  ProtocolType.Ethereum,
+  InterchainAccountConfig,
+  HyperlaneContracts<InterchainAccountFactories>
```
Comment (yorhodes) [inline typescript/sdk/src/core/EvmIcaModule.ts:18]: imo this should be `HyperlaneAddresses` instead of contracts?

Comment (yorhodes) [reply typescript/sdk/src/core/EvmIcaModule.ts:18]: or the constructor should take addresses instead of contracts

**Review by yorhodes — APPROVED** [review]
lgtm

File: typescript/sdk/src/core/EvmIcaModule.ts (hunk 2 above)
Comment (paulbalaji) [reply typescript/sdk/src/core/EvmIcaModule.ts:18]: yup that's correct - need to update the ISM module to do this too

Comment (ltyu) [reply typescript/sdk/src/core/EvmIcaModule.ts]: Done!

File: typescript/sdk/src/core/EvmIcaModule.hardhat-test.ts (hunk 1 above)
Comment (ltyu) [reply typescript/sdk/src/core/EvmIcaModule.hardhat-test.ts]: Made it a little more specific via address zero check

**Pull request merged**